oeps/oep-0025-proc-incremental-improvements.rst

```diff
@@ -0,0 +1,247 @@
+development tasks.
+
+Motivation
+==========
+
+There are often relatively straightforward changes to the Open edX code base
+which prove difficult to accomplish due to the sheer amount of code involved.
+These are not major new features or complex refactorings of the code, but
+rather things like:
+
+* Support a new version of Python or Django
+* Replace all uses of certain deprecated code with its preferred replacement
+* Find all instances of a particular antipattern and fix them
+
+Such projects can involve changing hundreds of files and thousands of lines of
+code...but they are fundamentally composed of dozens of routine, localized
+changes which can be made independently and without any particular prior
+knowledge of how the Open edX code is structured.  It is very difficult for
+any one organization to prioritize tackling one of these projects by itself,
+but once broken down into very small pieces, the benefits of distributing
+them across the entire community become rapidly apparent.
+
+Specification
+=============
+
+The `Incremental Improvements`_ (INCR) JIRA project is home to small,
+straightforward development tasks which must be finished in order to achieve
+larger, long-term objectives of the Open edX community. All tickets in it
+should have the following characteristics:
+
+* **Quick to implement.** Ideally no more than 3 hours for a developer who is
+  already familiar with the contribution process to submit a reasonable pull
+  request.
+
+* **Small in scope.** Should affect no more than 10 files (preferably fewer),
+  and not require large amounts of new or refactored code. If a similar change
+  needs to be made to dozens or hundreds of files, there should be a separate
+  ticket for each small batch of files (and there should ideally be a script
+  to automate the change).  This keeps the pull requests easy to review so
+  feedback can be provided promptly.
+
+* **No prerequisites.** Each INCR ticket should be doable immediately, without
+  waiting for another change to be made.
+
+* **Not urgent.** While all INCR tickets are important, they do not have
+  pressing deadlines. A new developer should have time to become familiar with
+  the contribution process and the changes to be made, without feeling
+  pressured to submit something quickly.
+
+* **Requires no additional information or clarification.** The ticket
+  description should clearly describe exactly what needs to be done.
+
+* **Specifies the required skill(s).** There should be labels on the ticket
+  listing any development skills required beyond the core ones needed to
+  submit a pull request. Typical labels include "javascript", "python", "css",
+  etc.
+
+* **Does not require knowledge of the existing Open edX code.** All INCR
+  tickets should be implementable by a developer with the appropriate
+  fundamental skills, and described assuming little or no knowledge of the
+  existing code. It can be assumed that the developer has read enough
```

Comment:
Maybe say `existing edx codebase` for the clarity that @nasthagiri is looking for?